### ISSUE TITLE
Save training metadata to model outputs

### ISSUE THREAD

**Issue by space-nuko** [kohya-ss/sd-scripts#47]
Hi thanks so much for your work with the LoRA training, its been a blast

I was thinking it would be a good help if the training parameters used for a LoRA model were saved to the resulting `.pt` file. I might want to remember how I configured a model and right now I have to remember to write down the parameters myself every time. Adding the data automatically would also help if I receive a model from somewhere else and want to know how they trained it

Some examples of things I would find useful in this metadata

- Directory structure/number of images/list of concepts/repeats
- Epoch count
- Batches per epoch
- Regularization image count
- LR scheduler/warmup rate
- Training batch size
- Learning rates

Given that the output files are PyTorch models (`.pt`) they seem to just be `.zip` files, so maybe just putting the training parameters as a `.json` file inside would suffice. And the `.safetensors` format has a JSON header also

From the `additional_networks` extension this data could later be inspected from a new tab or similar

It is important that the data is embedded into the `.pt` file itself so it is retained if the model is distributed later

**Comment by space-nuko** [comment]
> I'm testing the PR, saving the metadata to .safetensors works well. But if I save the metadata to .ckpt, I got an error while loading that .ckpt.

Whoops you're right, I accidentally overwrote the .zip when saving it. But when saving the .json as if it were a .zip the safe model loaders I tried reject the file anyway. I think your solution is good for now. Please feel free to revert the change for .ckpts if it blocks anyone
